### PR TITLE
make column update validation exceptions consistent

### PR DESCRIPTION
**Pull request by chaudum**
when trying to update a column in a table with a single primary key the
exception message referred to "clustered-by", because single PK is also
the routing column.

now both updating a single PK column and updating a part of a compound
PK give the same exception message.

### CODE REVIEW

File: sql/src/main/java/io/crate/analyze/relations/DocTableRelation.java [hunk 1]
```diff
@@ -137,7 +139,7 @@ private void ensureColumnCanBeUpdated(ColumnIdent ci) {
         }
     }
 
-    private void ensureNotUpdated(ColumnIdent columnUpdated, ColumnIdent protectedColumnIdent, String errorMessage) {
+    private void ensureNotUpdated(ColumnIdent columnUpdated, ColumnIdent protectedColumnIdent, String errorMessage) throws ColumnValidationException {
```
Comment (seut) [inline sql/src/main/java/io/crate/analyze/relations/DocTableRelation.java]: only non-runtime exceptions must be declared

File: sql/src/test/java/io/crate/analyze/InsertFromValuesAnalyzerTest.java [hunk 2]
```diff
@@ -1011,28 +1011,35 @@ public void testOnDuplicateKeyUpdateOnObjectColumn() throws Exception {
...
         expectedException.expectMessage("Validation failed for id: Updating a primary key is not supported");
-       e.analyze("insert into nested_pk (id, o) values (1, {b=1}) on duplicate key update id = 10");
+        e.analyze("insert into nested_pk (id, o) values (1, {b=1}) on duplicate key update id = 10");
+    }
+
+    @Test
+    public void testUpdateOnSinglePrimaryKeyColumnShouldRaiseAnError() throws Exception {
+        expectedException.expect(ColumnValidationException.class);
+        expectedException.expectMessage("Validation failed for id: Updating a primary key is not supported");
+        e.analyze("insert into users (id) values (1) on duplicate key update id = 10");
```
Comment (seut) [inline sql/src/test/java/io/crate/analyze/InsertFromValuesAnalyzerTest.java]: why is that needed? afaik this case is covered by the new DocTableRelationTest.testEnsureNotUpdateExceptionMessagePrimaryKey

Comment (chaudum) [reply sql/src/test/java/io/crate/analyze/InsertFromValuesAnalyzerTest.java]: then also testUpdateOnPartitionedColumnShouldRaiseAnError is obsolete

File: sql/src/main/java/io/crate/analyze/relations/DocTableRelation.java (hunk 1 above)
Comment (chaudum) [reply sql/src/main/java/io/crate/analyze/relations/DocTableRelation.java]: sure, stupid me ...

File: sql/src/test/java/io/crate/analyze/InsertFromValuesAnalyzerTest.java (hunk 2 above)
Comment (matriv) [reply sql/src/test/java/io/crate/analyze/InsertFromValuesAnalyzerTest.java]: In my opinion it makes more sense to keep these tests here than the manual call of valiation in DocTableRelationTest

Comment (chaudum) [reply sql/src/test/java/io/crate/analyze/InsertFromValuesAnalyzerTest.java]: I prefer more encapsulated, explicit testing, but I'm ok to keep them in the analyzer test.

File: sql/src/test/java/io/crate/analyze/relations/DocTableRelationTest.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import io.crate.metadata.doc.DocSchemaInfo;
+import io.crate.metadata.doc.DocTableInfo;
+import io.crate.metadata.table.TestingTableInfo;
+import io.crate.test.integration.CrateUnitTest;
+import io.crate.types.DataTypes;
+
+
+public class DocTableRelationTest extends CrateUnitTest {
+
+    public void testEnsureNotUpdateExceptionMessagePrimaryKey() {
```
Comment (seut) [inline sql/src/test/java/io/crate/analyze/relations/DocTableRelationTest.java]: maybe rename to testUpdatingPrimaryKeyThrowsException (maybe also all others below following this pattern)
